examples/hosts/app-integration/external-data/src/view/debugView.tsx

```diff
@@ -142,18 +145,19 @@ const ExternalDataView: React.FC<IExternalDataViewProps> = (props: IExternalData
 
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 interface ISyncStatusViewProps {
+    fluidSync: boolean;
     unsynchronizedChangesCount: number;
     handleCountUpdate: (count: number) => void;
 }
 
 // TODO: Implement the statuses below
 const SyncStatusView: React.FC<ISyncStatusViewProps> = (props: ISyncStatusViewProps) => {
-    useEffect(() => { }, [props.unsynchronizedChangesCount]);
+    useEffect(() => { }, [props.unsynchronizedChangesCount, props.fluidSync]);
```

Comment:
Do we need this empty effect? Both of the dependencies are input props, so this component should re-run if either of the values changes, right?

Reply:
IIRC it will re-render if the parent component changes the props its sending. But otherwise this component or another somewhere else could add or remove values to these prop objects and it will not trigger a re render unless you have a useEffect hook like this or make it part of state using useState.

Reply:
Hmm. Those other scenarios sound like a mis-use of React. I'm not sure we need to defend against them here. Props objects are assumed to be treated as immutable, even if javascript won't prevent that.

Reply:
Yeah, from my understanding the useEffect hook will directly update the DOM every time the items in the array change. @alexvy86 thoughts?

Reply:
I think I'm the least versed with React here, but reading a bit of its docs I think I agree with @Josmithr. I don't see what the `useEffect()` call is doing that wouldn't also happen without it (and that doesn't seem like a misuse of the props).